### PR TITLE
[DOCS-4333][CI Visibility] Add doc for junit upload detailing XPath tags feature

### PR DESCRIPTION
**Pull request by AdrianLC**
### What does this PR do?

New section explaining new feature for custom tags with XPath syntax in junit upload command from CI Visibility.

### Motivation

We need to explain a new feature

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

File: content/en/continuous_integration/tests/junit_upload.md [hunk 1]
```diff
@@ -322,9 +329,109 @@ For mobile apps (Swift, Android):
...
+
+
+`/testcase/@attribute-name`
+: The XML attribute from `<testcase attribute-name="value">`.
+
+`/testcase/../@attribute-name`
+: The XML attribute from the parent `<testsuite attribute-name="value">` of the current `<testcase>`.
+
+`/testcase/..//property[@name='property-name']`
+: The `value` attribute from the `<property name="property-name" value="value">` inside the parent `<testsuite>` of current `<testcase>`.
```
Comment (liashenko) [inline content/en/continuous_integration/tests/junit_upload.md:348]:
```suggestion
: The `value` attribute from the `<property name="property-name" value="value">` inside the parent `<testsuite>` of the current `<testcase>`.
```

File: content/en/continuous_integration/tests/junit_upload.md [hunk 2]
```diff
@@ -322,9 +329,109 @@ For mobile apps (Swift, Android):
...
+  </testsuite>
+</testsuites>
+{{< /code-block >}}
+
+{{< code-block lang="bash" >}}
+datadog-ci junit upload --service service_name \
+  --xpath-tag custom_tag=/testcase/..//property[@name=\'prop\'] ./junit.xml
+{{< /code-block >}}
+
+**Note:** the escaping of quotes with `\'`. The quotes are required and bash removes them if not escaped.
```
Comment (liashenko) [inline content/en/continuous_integration/tests/junit_upload.md:427]: Not sure it's the best way to put it, but the current is a bit difficult to understand.
```suggestion
**Note:** the quotes are required. Also, you have to escape quotes with slashes, e.g. `\'`.
```

**Review by liashenko — APPROVED** [review]

**Comment by buraizu** [comment]
Thanks, created DOCS-4333 to track as an editorial review

File: content/en/continuous_integration/tests/junit_upload.md [hunk 3]
```diff
@@ -322,9 +329,109 @@ For mobile apps (Swift, Android):
...
+
+`/testcase//property[@name='property-name']`
+: The `value` attribute from the `<property name="property-name" value="value">` inside the current `<testcase>`.
+
+Examples:
+
+{{< tabs >}}
+
+{{% tab "Test suite from @classname" %}}
+By default the `test.suite` tag of the tests is read from `<testsuite name="suite name">`. However some plugins might report a better value in `<testcase classname="TestSuite">`.
```
Comment (jtappa) [inline content/en/continuous_integration/tests/junit_upload.md:358]:
```suggestion
By default, the `test.suite` tag of the tests is read from `<testsuite name="suite name">`. However, some plugins might report a better value in `<testcase classname="TestSuite">`.
```

File: content/en/continuous_integration/tests/junit_upload.md [hunk 4]
```diff
@@ -322,9 +329,109 @@ For mobile apps (Swift, Android):
...
+
+{{< code-block lang="bash" >}}
+datadog-ci junit upload --service service_name \
+  --xpath-tag test.suite=/testcase/@aclassname ./junit.xml
+{{< /code-block >}}
+
+{{% /tab %}}
+
+{{% tab "Tag from attribute" %}}
+To add `custom_tag` to each test with values `value 1`, `value 2`.
```
Comment (jtappa) [inline content/en/continuous_integration/tests/junit_upload.md:382]:
```suggestion
To add `custom_tag` to each test with values `value 1`, `value 2`:
```

File: content/en/continuous_integration/tests/junit_upload.md [hunk 5]
```diff
@@ -322,9 +329,109 @@ For mobile apps (Swift, Android):
...
+
+{{< code-block lang="bash" >}}
+datadog-ci junit upload --service service_name \
+  --xpath-tag custom_tag=/testcase/@attr ./junit.xml
+{{< /code-block >}}
+
+{{% /tab %}}
+
+{{% tab "Tag from testsuite property" %}}
+To add `custom_tag` to each test with values `value 1`, `value 2`.
```
Comment (jtappa) [inline content/en/continuous_integration/tests/junit_upload.md:402]:
```suggestion
To add `custom_tag` to each test with values `value 1`, `value 2`:
```

File: content/en/continuous_integration/tests/junit_upload.md (hunk 2 above)
Comment (jtappa) [inline content/en/continuous_integration/tests/junit_upload.md:427]:
```suggestion
**Note:** The name must be in quotes. Bash requires quotes to be escaped using a backslash, for example `[@name='prop']` must be entered as `[@name=\'prop\'].
```

File: content/en/continuous_integration/tests/junit_upload.md [hunk 6]
```diff
@@ -322,9 +329,109 @@ For mobile apps (Swift, Android):
 **Examples**: `iPhone 12 Pro Simulator`, `iPhone 13 (QA team)`
 
 
+## Providing metadata with XPath expressions
+
+In addition to the `--tags` CLI parameter and the `DD_TAGS` environment variable, which apply custom tags globally to all tests included the uploaded XML report, the `--xpath-tag` parameter provides custom rules to add tags from different attributes within the XML to each test.
+
+The parameter provided must have the format `key=expression`, where `key` is the name of the custom tag to be added and `expression` is a valid [XPath][10] expression within the ones supported.
+
+XPath syntax is used for familiarity but only a limited subset of expressions are supported:
```
Comment (jtappa) [inline content/en/continuous_integration/tests/junit_upload.md:338]:
```suggestion
While XPath syntax is used for familiarity, but only the following expressions are supported:
```

**Review by jtappa — CHANGES_REQUESTED** [review]

File: content/en/continuous_integration/tests/junit_upload.md (hunk 6 above)
Comment (AdrianLC) [reply content/en/continuous_integration/tests/junit_upload.md:338]: With your suggestion we also need to remove "but" after the comma, right?
```
While XPath syntax is used for familiarity, only the following expressions are supported
```

Comment (jtappa) [reply content/en/continuous_integration/tests/junit_upload.md:338]: yes, good catch!

**Review by jtappa — APPROVED** [review]

**Pull request merged**